Cassity_Zach/Ruby_assignments/MYEnumerable.rb

```diff
@@ -0,0 +1,11 @@
+module MyEnumerable
+  def my_each
+    for i in 0..self.length-1 #self.length the array passed in below
+      puts self[i] #self refering to whatever number i is on
+    end
+  end
+end
+class Array
+   include MyEnumerable
+end
+[1,2,3,4].my_each { |i| puts i } #prints 1 2 3 4 in the terminal

```

Comment:
Despite the answer sheet, the purpose of this assignment is to understand blocks and yield.  Your implementation produces the same results no matter what is put in the block.  Try something like this...
```ruby
module MyEnumerable
  def my_each
    for i in self
      yield i
    end
  end
end
class Array
   include MyEnumerable
end
arr.my_each { |i| puts i }
```